File: leetcode/golang/45.跳跃游戏II.go
func jump(nums []int) int {
    goal := len(nums) - 1;
    ret := 0;
    for goal > 0 {
        for i, num := range nums {
            if num + i >= goal {
                goal = i;
                ret += 1;
                break;
            }
        }
    }
    return ret;
}

File: leetcode/python/45.跳跃游戏II.py
class Solution:
    def jump(self, nums: List[int]) -> bool:
        """ 贪心算法 """
        goal = len(nums) - 1
        ret = 0
        while goal > 0:
            for i in range(goal):
                if nums[i] + i >= goal:
                    goal = i
                    ret += 1
                    break
        return ret